pkg/rollup/config.go

```diff
@@ -0,0 +1,82 @@
+package rollup
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+
+	"github.com/timescale/promscale/pkg/internal/day"
+	"github.com/timescale/promscale/pkg/log"
+)
+
+type Config struct {
+	Enable               *bool `yaml:"enabled,omitempty"`
+	UseDefaultResolution *bool `yaml:"use_default_resolution"`
+	Resolution           `yaml:"resolutions,omitempty"`
+}
+
+type Info struct {
+	Resolution day.Duration `yaml:"resolution"`
+	Retention  day.Duration `yaml:"retention"`
+	Delete     bool         `yaml:"delete"`
+}
+
+type (
+	Label      string
+	Resolution map[Label]Info
+)
+
+var (
+	setDefaultDownsampleStateSQL = "SELECT prom_api.set_automatic_downsample($1)"
+	defaultDownsampleState       = false
+	useDefaultResolution         = true
+	defaultResolution            = map[Label]Info{
+		"short": {
+			Resolution: day.Duration(5 * time.Minute),
+			Retention:  day.Duration(90 * 24 * time.Hour),
+		},
+		"long": {
+			Resolution: day.Duration(time.Hour),
+			Retention:  day.Duration(395 * 24 * time.Hour),
+		},
+	}
+)
+
+func (d *Config) Apply(conn *pgx.Conn) error {
+	d.applyDefaults()
+
+	log.Info("msg", fmt.Sprintf("Setting automatic metric downsample to %t", *d.Enable))
+	if _, err := conn.Exec(context.Background(), setDefaultDownsampleStateSQL, d.Enable); err != nil {
+		return err
+	}
+
+	if *d.Enable {
+		if *d.UseDefaultResolution {
+			// Apply defaults only when user has not applied "short" & "long".
+			if _, exists := d.Resolution["short"]; !exists {
```

Comment:
Why don't we `block` keys that we use so this can't happen? I think it will make it easier to reason about rollups since we will always know that these two (`short` and `long`) are system rollups.

Reply:
I like this idea.